frontend/src/components/Common/ReportDialog/reportDoc.tsx

```diff
@@ -109,63 +114,40 @@ const ReportDoc = memo(
     }, [tableData, tableRowsNum, tableShowTotal]);
 
     const areasLegendDefinition: PDFLegendDefinition[] = useMemo(() => {
-      return [
-        {
-          value: 'Province',
-          style: [styles.dash, { backgroundColor: '#000000' }],
-        },
-        {
-          value: 'District',
-          style: [styles.dash, { backgroundColor: '#999797' }],
-        },
-        {
-          value: 'Township',
-          style: [styles.dash, { backgroundColor: '#D8D6D6' }],
-        },
-      ];
-    }, [styles.dash]);
+      return reportConfig.areasLegendDefinition.items.map(areaDefinition => {
+        return {
+          value: t(areaDefinition.title),
+          style: [styles.dash, { backgroundColor: areaDefinition.color }],
+        };
+      });
+    }, [reportConfig.areasLegendDefinition.items, styles.dash, t]);
 
-    const stormWindBuffersLegendDefinition: PDFLegendDefinition[] = useMemo(() => {
-      return [
-        {
-          value: 'Uncertainty Cones',
-          style: [
-            styles.borderedBox,
-            { backgroundColor: '#ffffff', borderColor: '#b8b1b1' },
-          ],
-        },
-        {
-          value: 'Wind Buffer 60 km/h',
-          style: [
-            styles.borderedBox,
-            { backgroundColor: '#fffcf1', borderColor: '#f7e705' },
-          ],
-        },
-        {
-          value: 'Wind Buffer 90 km/h',
-          style: [
-            styles.borderedBox,
-            { backgroundColor: '#ffeed8', borderColor: '#f99408' },
-          ],
+    const typeLegendDefinition: PDFLegendDefinition[] = useMemo(() => {
+      return reportConfig.typeLegendDefinition.items.map(
+        typeLegendDefinitionItem => {
+          return {
+            value: t(typeLegendDefinitionItem.title),
+            style: [
+              reportConfig.type === ReportTypeEnum.TROPICAL_STORMS
+                ? styles.borderedBox
+                : styles.box,
```

Comment:
This seems to be the only place where we use the `ReportTypeEnum`. Is there anyway to get rid of it altogether? Can you clarify what's switching here between the borderedBox and box styles?

Reply:
The `borderedBox` styles adds a border color and border width to the indication box of the legend Item in the legend section of the report. Practically we can remove the type at all from the configurations and just depend wether or not the type legend definition item has a border (which is an optional parameter). If it has we can add the `borderedBox` styling else we can add the `box` styling @ericboucher wdyt?